fe/src/main/java/org/apache/doris/common/proc/SchemaChangeProcNode.java

```diff
@@ -42,6 +58,110 @@ public SchemaChangeProcNode(SchemaChangeHandler schemaChangeHandler, Database db
         this.db = db;
     }
 
+    boolean filterResult(String columnName, Comparable element, HashMap<String, Expr> filter) throws AnalysisException {
+        if (filter == null) {
+            return true;
+        }
+        Expr subExpr = filter.get(columnName.toLowerCase());
+        if (subExpr == null) {
+            return true;
+        }
+        BinaryPredicate binaryPredicate = (BinaryPredicate) subExpr;
+        if (subExpr.getChild(1) instanceof StringLiteral && binaryPredicate.getOp() == BinaryPredicate.Operator.EQ) {
+            LOG.info("SubExpr value " + ((StringLiteral) subExpr.getChild(1)).getValue() + ", real value " + element);
+            return ((StringLiteral) subExpr.getChild(1)).getValue().equals(element);
+        }
+        if (subExpr.getChild(1) instanceof DateLiteral) {
+            Long leftVal = (new DateLiteral((String) element, Type.DATETIME)).getLongValue();
+            Long rightVal = ((DateLiteral) subExpr.getChild(1)).getLongValue();
+            LOG.info("Left value " + leftVal + ", right value " + rightVal);
+            switch( binaryPredicate.getOp()) {
+                case EQ:
+                case EQ_FOR_NULL:
+                    return leftVal == rightVal;
+                case GE:
+                    return leftVal >= rightVal;
+                case GT:
+                    return leftVal > rightVal;
+                case LE:
+                    return leftVal <= rightVal;
+                case LT:
+                    return leftVal < rightVal;
+                case NE:
+                    return leftVal != rightVal;
+                default:
+                    Preconditions.checkState(false, "No defined binary operator.");
+            }
+        }
+        return true;
+    }
+
+    public ProcResult fetchResultByFilter(HashMap<String, Expr> filter, ArrayList<OrderByPair> orderByPairs,
+                                          LimitElement limitElement) throws AnalysisException {
+        Preconditions.checkNotNull(db);
+        Preconditions.checkNotNull(schemaChangeHandler);
+
+        List<List<Comparable>> schemaChangeJobInfos = schemaChangeHandler.getAlterJobInfosByDb(db);
+
+        LOG.info("Begin fetch data.Filter size " + filter.size() );
+        for(Entry<String, Expr> entry : filter.entrySet()){
+            LOG.info("key " + entry.getKey() + ", value" + entry.getValue().getChild(1));
+        }
+
+        //where
+        List<List<Comparable>> jobInfos = new ArrayList<List<Comparable>>();
+        for (List<Comparable> infoStr : schemaChangeJobInfos) {
```

Comment:
if filter is null, jobInfos can be assigned with schemaChangeJobInfos without iterating schemaChangeJobInfos one by one